[quant][fx] Scope support for call_function in QuantizationTracer

supriyar opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #51259 [quant][fx] Update name of packed weight attributes
* #51171 [quant][fx] Make scale, zero_point buffers in the model, use FQN (for quantize_per_tensor ops)
* #51166 [quant][fx] Make scale, zero_point buffers in the model and use FQN (for quantized ops)
* **#51086 [quant][fx] Scope support for call_function in QuantizationTracer**

Summary:
Previously we only supported getting scope for call_module and custom qconfig dict for call_module.
This PR extends the Scope class to record the scope for all node types.
For call_function qconfig if module_name is specified it takes precedence over function qconfig.

Test Plan:
python test/test_quantization.py test_qconfig_for_call_func

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D26077602](https://our.internmc.facebook.com/intern/diff/D26077602)

supriyar committed:
[quant][fx] Extend scope to support all node type
Summary:
Previously we only supported call_module, but we need to get scope
for functional ops too

Test Plan:
python test/test_quantization.py test_qconfig_for_call_func

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

supriyar committed:
Update on "[quant][fx] Extend scope to support all node type"


Summary:
Previously we only supported call_module, but we need to get scope
for functional ops too

Test Plan:
python test/test_quantization.py test_qconfig_for_call_func

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D26077602](https://our.internmc.facebook.com/intern/diff/D26077602)

[ghstack-poisoned]

jerryzh168 reviewed (approved):
Please update the PR title and message as well

supriyar committed:
Update on "[quant][fx] Scope support for call_function in QuantizationTracer"


Summary:
Previously we only supported getting scope for call_module and custom qconfig dict for call_module.
This PR extends the Scope class to record the scope for all node types.
For call_function qconfig if module_name is specified it takes precedence over function qconfig.

Test Plan:
python test/test_quantization.py test_qconfig_for_call_func

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D26077602](https://our.internmc.facebook.com/intern/diff/D26077602)

[ghstack-poisoned]

supriyar committed:
Update on "[quant][fx] Scope support for call_function in QuantizationTracer"


Summary:
Previously we only supported getting scope for call_module and custom qconfig dict for call_module.
This PR extends the Scope class to record the scope for all node types.
For call_function qconfig if module_name is specified it takes precedence over function qconfig.

Test Plan:
python test/test_quantization.py test_qconfig_for_call_func

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D26077602](https://our.internmc.facebook.com/intern/diff/D26077602)

[ghstack-poisoned]
